### PR TITLE
Add new data, service and route for updating user info on front end

### PR DESCRIPTION
**Pull request by danachen**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|     | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### CODE REVIEW

File: src/routes/users.ts [hunk 1]
```diff
@@ -83,5 +83,19 @@ userRouter.put('/set-languages', getUserFromToken, async (req, res) => {
   return res.json(updatedUser);
 });
 
+userRouter.put('/update', getUserFromToken, async (req, res) => {
+  // const { user } = res.locals;
+  // console.log(user);
+  const { userId, userName, email } = req.body;
```
Comment (mrchrmn) [inline src/routes/users.ts:89]: The user id is never sent from the front end. That's what the token is for. You will have to use the `user.id` on `res.locals`.

**Review by mrchrmn — CHANGES_REQUESTED** [review]
Looks good except for the one thing I commented on.

File: src/services/users.ts [hunk 2]
```diff
@@ -127,6 +127,20 @@ const setUserLanguages = async function(
   return sanitizeUser(updatedUser);
 };
 
+const updateUserInfo = async function(
+  userId: string,
+  userName: string,
+  email: string,
+): Promise<SanitizedUser> {
+  const result = await userData.updateUserInfo(userId, userName, email);
+  console.log(result);
```
Comment (mrchrmn) [inline src/services/users.ts:136]: The console.log has to go eventually.

File: src/routes/users.ts (hunk 1 above)
Comment (danachen) [reply src/routes/users.ts:89]: True, I tried doing that, it was giving me the wrong user.id though.

**Review by mrchrmn — APPROVED** [review]

**Pull request merged**